examples/go/go.mod

```diff
@@ -0,0 +1,7 @@
+module provider-xyz-native
+
+go 1.20
+
+require (
+	github.com/pulumi/pulumi/sdk/v3 v3.30.0
```

Comment:
This is an ancient version of pulumi. Can we use `v3.94.1`?

Also, we are missing our provider: `replace github.com/pulumi/pulumi-xyz/sdk/go/xyz => ../../sdk/go/xyz` (or something similar).

Can you please run `go mod tidy && go mod build ./...` to verify that this mod file works.

Reply:
per slack convo, leaving as is & opening issue in pulumi/pulumi to resolve yaml2go code gen issue.